lib/dictionary.js

```diff
@@ -0,0 +1,76 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+"use strict";
+
+function Dictionary(dictionary) {
+    return new Proxy(dictionary, {
+        getPrototypeOf: function (_) {
+            return Dictionary.prototype;
+        },
+
+        get: function (target, key) {
+            if (typeof(target[key]) === "function") {
```

Comment:
Will this prevent overriding `put` and `remove` ?

Reply:
Yes, and `addListener`, etc.